src/main/java/uk/ac/ebi/subs/validator/core/validators/ReferenceValidator.java

```diff
@@ -0,0 +1,135 @@
+public class ReferenceValidator {
+
+    private final String FAIL_MESSAGE = "Could not find reference target: %s";
+    private final String SUCCESS_MESSAGE = "Reference found.";
+
+    @Autowired
+    public StudyRepository studyRepository;
+    @Autowired
+    public AssayRepository assayRepository;
+    @Autowired
+    public SampleRepository sampleRepository;
+
+    /**
+     * An Assay refers to a Study via a StudyRef.
+     * @param studyRef
+     * @param singleValidationResult
+     */
+    public void validate(StudyRef studyRef, SingleValidationResult singleValidationResult) {
+        Study study = studyRepository.findFirstByAccessionOrderByCreatedDateDesc(studyRef.getAccession());
+        updateSingleValidationResult(study, studyRef, singleValidationResult);
+    }
+
+    /**
+     * An AssayData refers to an Assay via an AssayRef
+     * @param assayRef
+     * @param singleValidationResult
+     */
+    public void validate(AssayRef assayRef, SingleValidationResult singleValidationResult) {
+        Assay assay = assayRepository.findFirstByAccessionOrderByCreatedDateDesc(assayRef.getAccession());
+        updateSingleValidationResult(assay, assayRef, singleValidationResult);
+    }
+
+    /**
+     * An AssayData refers to a Sample via a SampleRef
+     * @param sampleRef
+     * @param singleValidationResult
+     */
+    public void validate(SampleRef sampleRef, SingleValidationResult singleValidationResult) {
+        Sample sample = sampleRepository.findByAccession(sampleRef.getAccession());
+        updateSingleValidationResult(sample, sampleRef, singleValidationResult);
+    }
+
+    /**
+     * A Sample refers to other samples and itself via SampleRelationships
+     * @param sampleRelationshipList
+     * @param singleValidationResult
+     * @return
+     */
+    public SingleValidationResult validateSampleRelationships(List<SampleRelationship> sampleRelationshipList, SingleValidationResult singleValidationResult) {
+        StringBuilder accessions = new StringBuilder();
+
+        for (SampleRelationship sampleRelationship : sampleRelationshipList) {
+            Sample sample = sampleRepository.findByAccession(sampleRelationship.getAccession());
+
+            if (sample == null) {
+                if(accessions.toString().isEmpty()) {
+                    accessions.append(sampleRelationship.getAccession());
+                } else {
+                    accessions.append(", " + sampleRelationship.getAccession());
+                }
+            }
+
+            updateSingleValidationResult(accessions, singleValidationResult);
+        }
+        return singleValidationResult;
+    }
+
+    /**
+     * An Assay refers to multiple Samples via SampleUses
+     * @param sampleUseList
+     * @param singleValidationResult
+     */
+    public void validateSampleUses(List<SampleUse> sampleUseList, SingleValidationResult singleValidationResult) {
```

Comment:
There are code duplications in the `validateSampleRelationships` and `validateSampleUses` methods.
You can extract code inside the for loop into a private method with an accession parameter.